Tweaks to Bot Template

JBraunsmaJr opened the pull request:
1. Removed Net5 framework selection/support as Net6 is now released. 
2. Removed installation documentation, when we have a ready to use template we can add it back
3. Adjusted file structure to utilize file-scoped namespace
4. Updated Nuspec name
5. Updated Git URL references

Lulalaby commented on src/DisCatSharpTemplates.nuspec:
```diff
@@ -1,7 +1,7 @@
   <metadata>
-    <id>DisCatSharp.Hosting.ProjectTemplates</id>
+    <id>DisCatSharp.ProjectTemplates</id>
     <version>1.0.01</version>
```
```suggestion
    <version>1.0.2</version>
```

JBraunsmaJr replied:
fixed

JBraunsmaJr commented:
Add parameter for bot token for the bot template

![image](https://user-images.githubusercontent.com/28987352/142559766-63267126-8f9b-42c3-945d-d48d020bb1e3.png)

JBraunsmaJr commented:
Given the scenario multiple bot projects exist in a solution:

Web Host Template has an added task which copies all JSON files within the solution into the output directory.
This ensures all JSON files required for the bot(s) to run are included

Lulalaby commented:
❤️

JBraunsmaJr commented:
As noted in discord

Added solution template which gets updated when you run build-nuget.ps1

Updated individual project templates naming conventions to make it easier to aggregate

Added classifications to make template discovery easier

**We will have to update this project to point to the next stable release**

Lulalaby commented:
Okidoki

Lulalaby commented:
Bumped to 9.8.4 already for the release

JBraunsmaJr commented:
Added git-tagging support so we can tag via github which will represent the version number that'll get produced via build-nuget.ps1

It is important to note that the tags can only contain digits and decimal/periods 

9.8.4 is valid
9.8.4-rc1, for instance is not

Lulalaby commented:
That's nice!

JBraunsmaJr commented:
Apparently I don't have the access I used to have on this repo, but regardless I don't have the info for your nuget stuff anyhow

IDK what you want the keyname to be but inside of the SECRETS portion of the repo settings an API key needs to be set, it gets loaded into the workflow via ```${{ secrets.NUGET_API_KEY }}```

It's untested but in theory this should work as the other repo did something similar. Only thing I'm worried about is if the powershell stuff works first go through

Lulalaby commented:
Api key is set org wide

Lulalaby commented:
![image](https://user-images.githubusercontent.com/14029133/142716428-3ed375c5-d0fd-4cd8-88bc-10a3041e2203.png)
This should work?

JBraunsmaJr commented:
Didn't know there was an org one.  That will work though.

Lulalaby commented:
I keep most secrets org-wide, because we need them over and over again 
![:doggolul:](https://cdn.discordapp.com/emojis/866016686268612638.png?size=96)
